File: flipt-client-java/src/main/java/io/flipt/client/models/VariantEvaluationResponse.java
package io.flipt.client.models;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.util.List;
import lombok.Builder;
import lombok.Singular;
import lombok.Value;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Value
@Builder
@JsonInclude(JsonInclude.Include.NON_EMPTY)
@JsonDeserialize(builder = VariantEvaluationResponse.VariantEvaluationResponseBuilder.class)
@JsonIgnoreProperties(ignoreUnknown = true)
public class VariantEvaluationResponse {
  @JsonProperty("match")
  boolean match;

  @JsonProperty("segment_keys")
  @Singular
  List<String> segmentKeys;

  @JsonProperty("reason")
  String reason;

  @JsonProperty("flag_key")
  String flagKey;

  @JsonProperty("variant_key")
  String variantKey;

  @JsonProperty("variant_attachment")
  String variantAttachment;

  @JsonProperty("request_duration_millis")
  float requestDurationMillis;

  @JsonProperty("timestamp")
  String timestamp;

  @lombok.experimental.SuperBuilder
  @JsonIgnoreProperties(ignoreUnknown = true)
  public static class VariantEvaluationResponseBuilder {}
}

File: flipt-client-kotlin-android/src/main/java/io/flipt/client/models/VariantEvaluationResponse.kt
package io.flipt.client.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class VariantEvaluationResponse(
    @SerialName("match")val match: Boolean? = null,
    @SerialName("segment_keys")val segmentKeys: List<String>? = null,
    @SerialName("reason")val reason: String? = null,
    @SerialName("flag_key")val flagKey: String? = null,
    @SerialName("variant_key")val variantKey: String? = null,
    @SerialName("variant_attachment")val variantAttachment: String? = null,
    @SerialName("request_duration_millis")val requestDurationMillis: Float? = null,
    @SerialName("timestamp")val timestamp: String? = null
)